File: LeetCode/Python/1446. Consecutive Characters.py
class Solution:
    def maxPower(self, s: str) -> int:
        count = 0
        _max = 0
        at_hand = 'A'
        
        for char in s:
            if char == at_hand:
                count += 1
            else:
                at_hand = char
                count = 1
            _max = max(_max, count)
        
        return _max

File: LeetCode/Java/1446. Consecutive Characters.java
class Solution {
    public int maxPower(String s) {
        int n = s.length();
        int max = 1;
        int count = 0;
        HashSet<Character> set = new HashSet<>();
        
        for (int i = 0; i < n; i++) {
            char c = s.charAt(i);
            
            if (set.contains(c)) {
                count++;
                max = Math.max(count, max);
            }
            else {
                set = new HashSet<>();
                set.add(c);
                count = 1;
            }
        }
        
        return max;
    }
}
